src/uwtools/resources/FV3Forecast.jsonschema

```diff
@@ -43,44 +38,180 @@
           "type": "string"
         },
         "executable": {
-          "format": "uri",
           "type": "string"
         },
-        "fd_ufs": {
-          "$ref": "#/$defs/updatable_config"
-        },
```

Comment:
I think it might be good to keep this one around, but given that we don't know a ton about it, I believe it can go into the vat of static files for now. If anyone requests the ability to update it, we can add it to the configurable file list at that time.

How does that sound?

A more general question: how do we want to handle the user staging those required files that require a specific name? Are they required entries in the `static` block?

Reply:
Let me see if I understand: Are you suggesting that we remove explicit support for `fd_ufs` as a standalone entry in UW FV3Forecast YAML and just let users specify it, for now at least, in the `static:` block? If so, that sounds good to me.

For the more general question, I think you're asking whether we should explicitly require certain static files that we know must be present. That brings more questions to mind for me:

1. Is there a meaningful difference, from a driver's perspective, between "cycle specific" and "static" files, excep that the former may contain e.g. Jinja2 expressions related to the cycle that need to be rendered to select the correct files? Since a driver (or, at least, the FV3 driver we have so far) is specific to a given cycle, could we simplify and just have e.g. `files_to_link:`, which might contain both plain path strings (static files) and path strings with Jinja2 expressions, which may be cycle-specific (or could be based on something else in the config). Is the distinction useful enough to justify the complexity?
2. `uwtools` currently allows, for these file lists, either key-value pairs where the key is the symlink name relative to the run directory and the value is the target path, OR pairs where the key is a directory relative to the run directory, and the value is a list of target files. If we want to enforce that some files are _required_, it's definitely going to lead to a lot of duplication (and schema complexity) to enforce the same requirements for both styles of value -- and ideally it would be forbidden to specify the same run-directory file two different ways in the same config. I'm not even sure it's possible to enforce required elements in the second, list (JSON Schema `array`) style: If so, it may be through so-called "tuple validation", which allows for required and addition entries, but probably enforces _order_ of entries, too, and it might be a big ask from users to not only specify all these files, but keep them in a specific order. It would be a lot simpler to just have the first key-value style, as in a JSON Schema it's simple (not to mention DRY) to specify required keys and then permit additional, unknown, free-form key-value pairs. So: Is this list feature worth the complexity?
3. Given the above, we could potentially simplify and also support required link-into-run-directory files as well as optional, user-specified ones. How realistic is the use case where a user has changed the code such that some file `uwtools` thinks is required no longer is, but the schema forces the user to specify it anyway? I do think that it could be a big help to most users to tell them when they're not providing all the files we know they're going to need. Would we just help my hypothetical power user tweak their `FV3Forecast.jsonschema` to let them avoid specifying files they don't want? Suggest they just provide the file anyway, even though they don't need it?

Reply:
I think we sorted this out in group today: We'll now have `files_to_copy` and `files_to_link` YAML blocks and, in all cases, these blocks will contain `str` -> `str` key-value pairs where the keys are filenames/paths relative to the run directory, and values are absolute paths to files to copy/link. Keys and values may contain Python template syntax, which we'll render in the driver when copying/linking the files. (This is more likely to apply to to-be-linked files, which are more likely to be cycle-specific, and the formatting is likely to involve cycle-specific data.)

I think the schema is appropriately updated here. The copying/linking work will be handled in a different PR involving the graph-based driver work.